lib/analytics.js

```diff
@@ -212,17 +212,50 @@ Analytics.prototype.identify = function(id, traits, options, fn) {
   if (is.object(id)) options = traits, traits = id, id = user.id();
   /* eslint-enable no-unused-expressions, no-sequences */
 
+  var plan = this.options.plan || {};
+  var traitPlans = plan.identify || {};
+  var originalTraits = extend({}, traits);
+
+  for (var trait in traits) {
+    if (traits.hasOwnProperty(trait)) {
+      var traitPlan = traitPlans[trait];
+
+      if (traitPlan) {
+        this.log('identify plan %o - %o', trait, traitPlan);
+        if (traitPlan.enabled === false) {
+          delete traits[trait];
```

Comment:
I think we should clone the traits and delete from the copy, rather than mutating what the user passes in.